core/corehttp/gateway.go

```diff
@@ -156,7 +156,7 @@ func (o *offlineGatewayErrWrapper) ResolvePath(ctx context.Context, path gateway
 	return md, err
 }
 
-func (o *offlineGatewayErrWrapper) GetCAR(ctx context.Context, path gateway.ImmutablePath, params gateway.CarParams) (gateway.ContentPathMetadata, io.ReadCloser, error) {
+func (o *offlineGatewayErrWrapper) GetCAR(ctx context.Context, path gateway.ImmutablePath, params *gateway.CarParams) (gateway.ContentPathMetadata, io.ReadCloser, error) {
```

Comment:
You should add an interface guard.

Reply:
@Jorropo hmm, how would that benefit this? The way it is done right now, the code will fail at compile time already.

Reply:
Ah yes NVM thought the interface definition would be here.
Btw you need to be carefull when passing pointers to interfaces, they almost always allocate (they more than likely here).